lxd/device/device.go

```diff
@@ -0,0 +1,115 @@
+package device
+
+import (
+	"fmt"
+	"github.com/lxc/lxd/lxd/state"
+)
+
+// ErrUnsupportedDevType is the error that occurs when an unsupported device type is created.
+var ErrUnsupportedDevType = fmt.Errorf("Unsupported device type")
+
+// Config represents user defined persistent config for a device.
+type Config map[string]string
+
+// VolatileConfig represents LXD defined persistent config for a device.
+type VolatileConfig map[string]string
+
+// RunConfig represents LXD defined run-time config used for device setup.
+type RunConfig map[string]string
```

Comment:
I suspect we'll want to rework that one a bit. Current key/value works well for network devices where you can just assume `lxc.net.ID.KEY=VALUE` but that may become a bit more limited if say you also wanted to set a `lxc.cgroup` entry at the same time or a `lxc.mount` entry.

One way is to do some careful namespacing of those keys, another would be to add more structure to it by returning something like a `RunConfigLXC` struct that could look something like:
```
type RunConfigLXC struct {
    NetworkInterfaces []map[string]string
    Mounts []map[string]string
    Cgroups []map[string]string
}
```

So we may be able to abstract things sufficiently that in `container_lxc.go` we just need one loop through all devices, call the deviceStart for them and not need any type-specific knowledge about them.

Reply:
@stgraber I will have a think about this, so I understand fully, can you advise when it would be needed for a nic device to set non-nic properties (like cgroup/mounts?), or are you meaning for other device types its more likely that combined settings would need to be applied for a single device?

Reply:
@stgraber I've changed RunConfig to be the struct as you suggested, I didn't call it RunConfigLXC as wanted to keep it generic (although we may not use all of them for different instance types, or they may have different keys).